qiskit/circuit/qpy_serialization.py

```diff
@@ -10,1913 +10,17 @@
-
-
-.. [#f1] https://tools.ietf.org/html/rfc1700
-.. [#f2] https://numpy.org/doc/stable/reference/generated/numpy.lib.format.html
-.. [#f3] https://docs.python.org/3/c-api/complex.html#c.Py_complex
-"""
-from collections import namedtuple
-import io
-import json
-import struct
-import uuid
 import warnings
 
-import numpy as np
-
-from qiskit.circuit.quantumcircuit import QuantumCircuit
-from qiskit.circuit.quantumregister import QuantumRegister, Qubit
-from qiskit.circuit.classicalregister import ClassicalRegister, Clbit
-from qiskit.circuit.parameter import Parameter
-from qiskit.circuit.parameterexpression import ParameterExpression
-from qiskit.circuit.parametervector import ParameterVector, ParameterVectorElement
-from qiskit.circuit.gate import Gate
-from qiskit.circuit.instruction import Instruction
-from qiskit.circuit import library
-from qiskit import circuit as circuit_mod
-from qiskit import extensions
-from qiskit.circuit import controlflow
-from qiskit.extensions import quantum_initializer
-from qiskit.version import __version__
-from qiskit.exceptions import QiskitError
-from qiskit.quantum_info.operators import SparsePauliOp
-from qiskit.synthesis import evolution as evo_synth
-from qiskit.utils import optionals as _optionals
-
-
-# v1 Binary Format
-# ----------------
-# FILE_HEADER
-FILE_HEADER = namedtuple(
-    "FILE_HEADER",
-    ["preface", "qpy_version", "major_version", "minor_version", "patch_version", "num_circuits"],
-)
-FILE_HEADER_PACK = "!6sBBBBQ"
-FILE_HEADER_SIZE = struct.calcsize(FILE_HEADER_PACK)
-
-# HEADER binary format
-HEADER_V2 = namedtuple(
-    "HEADER",
-    [
-        "name_size",
-        "global_phase_type",
-        "global_phase_size",
-        "num_qubits",
-        "num_clbits",
-        "metadata_size",
-        "num_registers",
-        "num_instructions",
-    ],
-)
-HEADER_V2_PACK = "!H1cHIIQIQ"
-HEADER_V2_SIZE = struct.calcsize(HEADER_V2_PACK)
-
-HEADER = namedtuple(
-    "HEADER",
-    [
-        "name_size",
-        "global_phase",
-        "num_qubits",
-        "num_clbits",
-        "metadata_size",
-        "num_registers",
-        "num_instructions",
-    ],
-)
-HEADER_PACK = "!HdIIQIQ"
-HEADER_SIZE = struct.calcsize(HEADER_PACK)
-
-# CUSTOM_DEFINITIONS
-# CUSTOM DEFINITION HEADER
-CUSTOM_DEFINITION_HEADER = namedtuple("CUSTOM_DEFINITION_HEADER", ["size"])
-CUSTOM_DEFINITION_HEADER_PACK = "!Q"
-CUSTOM_DEFINITION_HEADER_SIZE = struct.calcsize(CUSTOM_DEFINITION_HEADER_PACK)
-
-# CUSTOM_DEFINITION
-CUSTOM_DEFINITION = namedtuple(
-    "CUSTOM_DEFINITON",
-    ["gate_name_size", "type", "num_qubits", "num_clbits", "custom_definition", "size"],
+warnings.warn(
+    "Importing QPY serialization from qiskit.circuit.qpy_serialization "
+    "has been deprecated. Use new import path qiskit.qpy instead. "
+    "This import path will be removed after sufficient deprecation period from 0.20 release.",
+    ImportWarning,
```

Comment:
I apologize apparently this comment got lost in my earlier review.

We can't deprecate this quite yet. Per the updated (which was done in December last year, so it's a new requirement) deprecation policy https://qiskit.org/documentation/contributing_to_qiskit.html#deprecation-policy we need to release the alternative first before deprecation. That being said, I think an import warning here is ok because it's ignored by the default filters (at least according to: https://docs.python.org/3/library/exceptions.html#ImportWarning ). But we should probably drop the release note for 0.20.0 we can change this to a more active `DeprecationWarning` and have the release note in 0.21.0.

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/7582/commits/d011180be96f51fc8c7d2174ab9b4b2fcfdddc0f. The warning is replaced with TODO comment.